runtime/benches/accounts.rs

```diff
@@ -141,3 +142,57 @@ fn bench_delete_dependencies(bencher: &mut Bencher) {
         accounts.accounts_db.clean_accounts(None);
     });
 }
+
+#[bench]
+#[ignore]
+fn bench_concurrent_read_write(bencher: &mut Bencher) {
```

Comment:
btw, do you have any idea of how _single-threaded_ throughput of writing is changed from `std::collections::HashMap` to `DashMap`? I think I'm worrying too much but I'd rather want to confirm how far does `DashMap` do well while supporting the concurrency via sharding. Maybe it's trading off maximum throughput by negligible margin?

Reply:
To add more context, our basic tenet is _batch them if we can, make it concurrent otherwise_. so, I guess the upper layers slumming the AccountsDB optimized for batching and the single threaded perf is kind of moderately related to the batching perf. Thus, we're somewhat sensitive to it.

Reply:
@ryoqun that's a good point, I've run the benchmark here: https://github.com/xacrimon/conc-map-bench which has three different work profiles, exchange(read-write), cache (read-heavy), and rapid-grow (insert-heavy), more details can be found in that link above. The results for a single thread on my Macbook Pro: 

== cache
-- MutexStd
25165824 operations across 1 thread(s) in 14.941146454s; time/op = 593ns

-- DashMap
25165824 operations across 1 thread(s) in 15.589596263s; time/op = 619ns
==

== exchange
-- MutexStd
25165824 operations across 1 thread(s) in 20.954264682s; time/op = 831ns

-- DashMap
25165824 operations across 1 thread(s) in 20.875345754s; time/op = 828ns
==

== rapid grow
-- MutexStd
25165824 operations across 1 thread(s) in 20.22593938s; time/op = 802ns
==

-- DashMap
25165824 operations across 1 thread(s) in 17.456807471s; time/op = 693ns
==

Reply:
@ryoqun, good suggestions, here's the results I saw on my Macbook Pro:

`bench_concurrent_read_write` on 1 writer, no readers:`

```
DashMap:
test bench_concurrent_read_write  ... bench:   3,713,260 ns/iter (+/- 679,081)

Master: 
test bench_concurrent_read_write  ... bench:   3,773,731 ns/iter (+/- 654,643)
```

`accounts-bench/src/main.rs:`

```
DashMap:
clean: false
Creating 10000 accounts
created 10000 accounts in 4 slots create accounts took 148ms

Master: 
clean: false
Creating 10000 accounts
created 10000 accounts in 4 slots create accounts took 145ms
```

Reply:
@carllin Perfect reporting :)